include/veriblock/blockchain/base_block_tree.hpp

```diff
@@ -340,6 +342,14 @@ struct BaseBlockTree {
     return validity_sig_.disconnect(id);
   }
 
+  index_t& getRoot() {
+    VBK_ASSERT_MSG(isBootstrapped(), "must be bootstrapped");
+    return *getBestChain().first();
+  }
+
+  //! the number of blocks that have BLOCK_APPLIED flag set
+  block_height_t appliedBlockCount = 0;
```

Comment:
should it be public?

Reply:
either it has to be public since it's accessed by other classes, or we need to add blockApplied() blockUnapplied(), getAppliedBlockCount()